Common/app/mqtt/mqtt_agent_task.c

```diff
@@ -424,27 +425,33 @@ static inline void prvCompressCallbackList( SubCallbackElement_t * pxCallbackLis
             {
                 if( pxCallbackList[ uxLastOccupiedIndex ].pxSubInfo != NULL )
                 {
-                    /* Move new item into place */
+                    configASSERT( uxLastOccupiedIndex > uxIdx );
+
+                    /* Move context from uxLastOccupiedIndex to uxIdx */
```

Comment:
Should uxCallbackCount be incremented here as well?

Reply:
Yes. Missed that one.